### ISSUE TITLE
meta data is overwirtten with constant data

### ISSUE THREAD

**Issue by najtin** [detecttechnologies/thermal_base#6]
Hi, i just stumbled upon this project so i am not sure if i understand correctly. 

To me this seems wrong:
https://github.com/detecttechnologies/thermal_base/blob/main/thermal_base/thermal_base.py#L110-L125

I am going to try to built it myself and report back.

**Comment by najtin** [comment]
It seems like someone had at least partial success in reverse engineering the important meta data: https://exiftool.org/forum/index.php?topic=11401.0

**Comment by rsnk96** [comment]
Hey @najtin 

Yes absolutely, it would be better if we can extract the formula params from the metadata itself for the DJI encoding too. Would you be interested in contributing the same?

**Comment by najtin** [comment]
Yes i will take a look into that. Though i have no experience in reverse engineering and it would be a side project for me. :D

**Comment by rsnk96** [comment]
Haha well, all the best, hope you figure it out! 🙂

**Comment by omritz** [comment]
Hi, Regarding the same issue, DJI released [SDK ](https://www.dji.com/de/zenmuse-h20-series/downloads) that allows extracting the information from images taken by H20T and XTS but does not yet support M2EA. According to this [thread](https://forum.dji.com/forum.php?mod=viewthread&tid=230321), one of their developers says they will release a version that supports M2EA in the second half of 2021.  The formula that DJI used to calculate the temperatures in deg C is different than the one that you should use for FLIR data.

**Comment by rsnk96** [comment]
Thanks for sharing @omritz . Will look into integrating that!

**Comment by rsnk96** [comment]
Hey @omritz - the DJI SDK has been integrated by @agarwalkunal12 for fetching the raw sensor values 😃

**Comment by rsnk96** [comment]
Can you test it out, and share how it compares to using the DJI Thermal Analysis tool?

**Comment by omritz** [comment]
Thanks for the update. When I test it with the new M2EA support and adjust the same parameters, the results are not the same as the DJI thermal analysis tool.
For the same image:
DJI Thermal Analysis - max=55.3, min=16.8, avg=35.1
Thermal_base - max=38.2, min=26.7 , avg=32.2

**Comment by rsnk96** [comment]
@agarwalkunal12 - any thoughts on this? Could this be because of the assumptions made for the missing metadata?

**Comment by nrontsis** [comment]
Perhaps this could be because DJI's SDK [is not called](https://github.com/detecttechnologies/thermal_base/blob/main/thermal_base/thermal_base.py#L162-L183) when M2EA images are processed by `thermal_tools`?
